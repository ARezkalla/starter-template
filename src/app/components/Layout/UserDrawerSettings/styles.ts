export const styles = {
  drawerContainer: {
    paddingTop: 7,
    width: 280,
    margin: 1,
  },
}
