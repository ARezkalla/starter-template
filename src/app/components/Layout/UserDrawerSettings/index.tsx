import React from "react"

import { setIsSettingsOpen } from "@store/rootSlice"

import { Box, Drawer } from "@mui/material"

import { styles } from "./styles"
import { useAppDispatch, useAppSelector } from "@store/hooks"

interface IAppDrawerSettings {
  children: React.ReactNode
}

const UserDrawerSettings = (props: IAppDrawerSettings) => {
  const { isSettingsOpen } = useAppSelector((state) => state.global)
  const dispatch = useAppDispatch()

  const handleDrawerClose =
    () => (event: React.KeyboardEvent | React.MouseEvent) => {
      if (
        "type" in event &&
        event.type === "keydown" &&
        ((event as React.KeyboardEvent).key === "Tab" ||
          (event as React.KeyboardEvent).key === "Shift")
      ) {
        return
      }

      dispatch(setIsSettingsOpen(false))
    }

  return (
    <Drawer
      anchor="right"
      open={isSettingsOpen}
      variant="temporary"
      onClose={handleDrawerClose()}
    >
      <Box
        sx={styles.drawerContainer}
        role="presentation"
        onClick={handleDrawerClose()}
        onKeyDown={handleDrawerClose()}
      >
        {props.children}
      </Box>
    </Drawer>
  )
}

export default UserDrawerSettings
