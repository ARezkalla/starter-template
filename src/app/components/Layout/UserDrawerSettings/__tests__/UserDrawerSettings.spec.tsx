import { renderWithProviders } from "@/app/test/renderWithProvider"
import { expect, it, describe } from "vitest"
import UserDrawerSettings from "../"

describe("UserDrawerSettings", () => {
  it("should render without crashing", () => {
    const { container } = renderWithProviders(
      <UserDrawerSettings>
        <></>
      </UserDrawerSettings>,
    )
    expect(container).toMatchSnapshot()
  })
})
