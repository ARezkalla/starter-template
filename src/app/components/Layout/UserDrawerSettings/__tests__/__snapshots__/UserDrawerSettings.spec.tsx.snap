// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserDrawerSettings > should render without crashing 1`] = `<div />`;

exports[`should render without crashing 1`] = `<div />`;
